src/main/scala/temple/generate/server/go/service/GoServiceHookGenerator.scala

```diff
@@ -1,10 +1,56 @@
 package temple.generate.server.go.service
 
+import temple.generate.CRUD
+import temple.generate.CRUD.CRUD
 import temple.generate.server.ServiceRoot
+import temple.generate.server.go.common.GoCommonGenerator.{genStruct, genMethod, genReturn}
+import temple.generate.utils.CodeTerm.mkCode
 import temple.utils.StringUtils.doubleQuote
 
+import scala.collection.immutable.ListMap
+
 object GoServiceHookGenerator {
 
   private[service] def generateImports(root: ServiceRoot): String =
     s"import ${doubleQuote(root.module + "/dao")}"
+
+  private[service] def generateHookStruct(root: ServiceRoot, operations: Set[CRUD]): String = {
+    val beforeCreate = operations.toSeq.sorted.map {
+      case op @ (CRUD.List | CRUD.Read | CRUD.Delete) =>
```

Comment:
Was not aware of this syntax until now - pretty fancy

Reply:
neither was I, useful